src/linear_solver_routines/block_col_projection.jl

```diff
@@ -0,0 +1,64 @@
+# This file is part of RLinearAlgebra.jl
+"""
+    LinSysBlkColGent <: LinSysBlkColProjection
+
+A mutable structure that represents a standard block column projection method that projects 
+a previous iterate onto the columnspace of an inputted block. This method can be viewed as 
+a generalization to block coordinate descent. The updates to this are computed
+using Gentleman's Algorithm.
+
+For more information on Gentleman's see:
+Miller, Alan J. “Algorithm AS 274: Least Squares Routines to Supplement Those of Gentleman.” 
+Journal of the Royal Statistical Society. Series C (Applied Statistics), 
+vol. 41, no. 2, 1992, pp. 458–78. JSTOR, https://doi.org/10.2307/2347583. 
+
+# Aliases
+- `BlockCoordinateDescent`
+
+# Fields
+- `α::Float64`, a relaxation parameter that should be set between `0.0` and `2.0`
+- `gent::GentData`, a buffer to store important information related to the Gentleman's 
+incremental QR least squares solver.
+- `update::Union{Nothing, AbstractArray}`, a buffer for storing update.
+- `rowsize::Int64`, the upper limit on the size of the row blocks in Gentleman's. By default this is set to 10000.
+
+# Constructors
+Calling the constructor `LinSysBlkColProj()` defaults the relaxation parameter to `1.0`.
+"""
+mutable struct LinSysBlkColGent <: LinSysBlkColProjection 
+    α::Float64
+    gent::Union{Nothing, GentData}
+    update::Union{Nothing, AbstractArray}
+    rowsize::Int64
+end
+
+LinSysBlkColGent(;α = 1.0, rowsize = 10000) = LinSysBlkColGent(α, nothing, nothing, rowsize)
```

Comment:
Make this constructor with a warning if `rowsize` is larger than 1 GB then the user should know what resources their computer has.

Reply:
I have added this warning message.